samples/set_best_video_mode/main.cpp

```diff
@@ -0,0 +1,73 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+#include <vector>
+
+// Screen dimension globals
+static int SCREEN_WIDTH = 640;
+static int SCREEN_HEIGHT = 480;
+
+int main(void)
+{
+    // set the default video mode
+    XVideoSetMode(SCREEN_WIDTH, SCREEN_HEIGHT, 32, REFRESH_DEFAULT);
+
+    // create an empty list for the available video modes
+    std::vector<VIDEO_MODE> availableVideoModes;
+
+    // save the best video mode here
+    VIDEO_MODE bestVideoMode = {0, 0, 0, 0};
+
+    VIDEO_MODE vm;
+    int bpp = 32;  // Bits per pixel
+    void *p = NULL;  // Initialize to NULL for the first call
+
+    // get the available video modes
+    while (XVideoListModes(&vm, bpp, REFRESH_DEFAULT, &p)) {
+        availableVideoModes.push_back(vm);
+
+        // ensure height is equal to or better than the current best video mode
+        if (vm.height < bestVideoMode.height) {
+            continue;
+        }
+
+        // ensure width is equal to or better than the current best video mode
+        if (vm.width < bestVideoMode.width) {
+            continue;
+        }
+
+        // ensure bpp is equal to or better than the current best video mode
+        if (vm.bpp < bestVideoMode.bpp) {
+            continue;
+        }
+
+        // ensure refresh is equal to or better than the current best video mode
+        if (vm.refresh < bestVideoMode.refresh) {
+            continue;
+        }
+
+        // save the best video mode
+        bestVideoMode = vm;
+    }
+
+    SCREEN_WIDTH = bestVideoMode.width;
+    SCREEN_HEIGHT = bestVideoMode.height;
```

Comment:
nit: Use `bestVideoMode.width` and `bestVideoMode.height` directly in the call to `XVideoSetMode` below instead of reassigning the static variables to reduce the potential for errors when someone is reusing this code

Reply:
It's been a while since I did this, but I believe I wanted to set the variables in the outer scope in case they would be needed for another function call later. Doesn't matter for this example though, so I'm happy to adjust it.